src/Frame.cpp

```diff
@@ -818,16 +818,20 @@ void Frame::AddImage(std::shared_ptr<QImage> new_image, bool only_odd_lines)
 		// Ignore image of different sizes or formats
 		bool ret=false;
 		#pragma omp critical (AddImage)
-		if (image == new_image || image->size() != image->size() || image->format() != image->format())
-			ret=true;
-		if (ret)
-			return;
-
+		{
+			if (image == new_image || image->size() != new_image->size()) {
+				return;
+			}
+			if (new_image->format() != image->format()) {
+				new_image = std::shared_ptr<QImage)(new_image->convertToFormat(image->format()));
+			}
+		}
+		
 		// Get the frame's image
 		const GenericScopedLock<juce::CriticalSection> lock(addingImageSection);
 		#pragma omp critical (AddImage)
 		{
-			const unsigned char *pixels = image->constBits();
+			unsigned char *pixels = image->bits();
```

Comment:
`const`s are nice and all, when appropriate, but when you're modifying the pixels in the buffer I'm pretty sure we're not supposed to use `constBits()` since it suppresses Qt's copy-on-write handling for shared `QImage` buffers.